src/model_signing/_signing/pkcs11uri.py

```diff
@@ -0,0 +1,473 @@
+# Copyright (c) 2025, IBM CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functionality to parse PKCS #11 URIs and access its components."""
+
+from collections.abc import Iterable
+import glob
+import os
+from os import stat
+import re
+from stat import S_ISREG
+from typing import Optional
+from urllib import parse
+from urllib.parse import urlparse
+
+import PyKCS11
+
+
+def escape_all(s: str) -> str:
```

Comment:
Can you add docstrings to all functions, please? It's not clear what this does and why, for example.

Reply:
(sorry, missed this file until now, gitHub was showing it collapsed to me)

Reply:
Done.